bin/p2-exec/main.go

```diff
@@ -247,7 +274,46 @@ func cgEnter(platconf, launchableName, cgroupName string) error {
 	} else if err != nil {
 		return util.Errorf("Could not set cgroup parameters: %s", err)
 	}
-	return cg.AddPID(cgConfig.Name, 0)
+	return cg.AddPID(config.Name, 0)
+}
+
+// given ENV variable "configPath" return parameters for the new cgroup being created
+func cgGetConfig(configPath string, configKey string) (cgroups.Config, error) {
+	if config := os.Getenv(configPath); config != "" {
+		confBuf, err := ioutil.ReadFile(config)
+		if err != nil {
+			return cgroups.Config{}, err
+		}
+		if configPath == "RESOURCE_LIMITS" {
+			cgMap := make(map[string]cgroups.Config)
+			err = yaml.Unmarshal(confBuf, cgMap)
+			if err != nil {
+				return cgroups.Config{}, err
+			}
+			if _, ok := cgMap["cgroup"]; !ok {
+				return cgroups.Config{}, util.Errorf("Key %q in %s does not contain cgroup parameters\n", configKey, configPath)
+			}
+			cgConfig := cgMap["cgroup"]
+			return cgConfig, nil
+		} else {
+			cgMap := make(map[string]map[string]cgroups.Config)
+			err = yaml.Unmarshal(confBuf, cgMap)
+			if err != nil {
+				return cgroups.Config{}, err
+			}
+
+			if _, ok := cgMap[configKey]; !ok {
+				return cgroups.Config{}, util.Errorf("Key %q not found in %s", configKey, configPath)
+			}
+			if _, ok := cgMap[configKey]["cgroup"]; !ok {
+				return cgroups.Config{}, util.Errorf("Key %q in %s does not contain cgroup parameters\n", configKey, configPath)
+			}
+			cgConfig := cgMap[configKey]["cgroup"]
+			return cgConfig, nil
+		}
+	} else {
+		return cgroups.Config{}, util.Errorf("No %s found in environment", configPath)
```

Comment:
i'd put this at the top of the function inside a `if config == ""`, that will save you some indentation and save people reading this from having to scroll up to see what the other cases were